rustfmt.toml

```diff
@@ -1,6 +1,11 @@
 # configuration for https://rust-lang.github.io/rustfmt/
+comment_width = 100
+format_code_in_doc_comments = true
+format_macro_matchers=true
 group_imports = "StdExternalCrate"
 imports_granularity = "Crate"
+normalize_doc_attributes=true
+reorder_impl_items=true
+unstable_features=true
```

Comment:
I just read a bit more about the [unstable_features](https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#unstable_features) setting. Sounds like this makes it possible to run the unstable rustfmt from the stable compiler!!! But, it's required to pass this at the command line for it to work (https://github.com/rust-lang/rustfmt/issues/3387#issuecomment-1867606088 / https://github.com/rust-lang/rustfmt/issues/5511#issuecomment-1867600019 have the details).

I think perhaps add this to the makefile and ci.yml to make this part of the formatting / checks that happen there and we can drop one of the needs for nightly. Docs lint still requires it in order to scrape the examples. I wonder if we actually need that in CI, or if we could just compile all with stable?

Reply:
Did you want to do the CI / makefile changes before merging? Happy either way.

Reply:
I am thinking about removing the `unstable_feature` from the config file… Given that they are unstable its kind of correct to prefer nightly for this.

I am also curious about these things getting stablized. Some of them are also done by rust-analyzer so nothing completely experimental.

Reply:
I still think keeping unstable stuff on nightly seems the better thing to do. I wonder why this even exists as its either stable or not… this is not, but it's on the stable rust version… somewhat weird.